### PR TITLE
DT-1072: ✨ Map multi user and single user azuread

### PR DESCRIPTION
**Pull request by petergphillips**
At present will just work for the single user matched scenario.
Need to add some tests for the changes.

### CODE REVIEW

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/integration/ClientLoginSpecification.kt [hunk 1]
```diff
@@ -122,6 +122,12 @@ class ClientLoginSpecification : AbstractAuthSpecification() {
       .jsonPath(".user_id").isEqualTo("4")
       .jsonPath(".sub").isEqualTo("RO_USER")
       .jsonPath(".auth_source").isEqualTo("nomis")
+      .jsonPath(".access_token").value<JSONArray> {
+        val claims = JWTParser.parse(it[0].toString()).jwtClaimsSet
+        assertThat(claims.getClaim("user_name")).isEqualTo("RO_USER")
+        assertThat(claims.getClaim("authorities") as List<*>)
+          .containsExactly("ROLE_GLOBAL_SEARCH", "ROLE_PRISON", "ROLE_INACTIVE_BOOKINGS", "ROLE_LICENCE_RO")
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/digital/hmpps/oauth2server/integration/ClientLoginSpecification.kt:129]: we get the mapping of roles for free for the single user case with the new approach

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/JWTTokenEnhancer.java [hunk 2]
```diff
@@ -41,45 +37,32 @@
     @Autowired
     private JdbcClientDetailsService clientsDetailsService;
 
-    @Autowired
-    private UserContextService userContextService;
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/JWTTokenEnhancer.java:45]: revert all these changes, will be done earlier in the chain by the authorization endpoint instead

File: .editorconfig [hunk 3]
```diff
@@ -1,9 +1,16 @@
-[*.{kt,kts}]
```
Comment (petergphillips) [inline .editorconfig:1]: seemed to be struggling with this line, kept on reformatting the build file with 4 spaces instead.  Splitting out fixed it.

File: .editorconfig [hunk 4]
```diff
@@ -1,9 +1,16 @@
...
+[*.kt]
 indent_size = 2
-max_line_length = 120 
+max_line_length = none
```
Comment (petergphillips) [inline .editorconfig:3]: we weren't actually using 120, so corrected

File: .editorconfig [hunk 5]
```diff
@@ -1,9 +1,16 @@
...
 
 [*.java]
 ij_java_class_count_to_use_import_on_demand = 99
 ij_java_names_count_to_use_import_on_demand = 99
 ij_java_use_single_class_imports = true
 ij_java_insert_inner_class_imports = true
+
+[*.html]
+indent_size = 2
```
Comment (petergphillips) [inline .editorconfig:16]: sort out web page format too

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/UserSelectorAuthorizationEndpoint.kt [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+    val replacedAuthentication = if (!account.isNullOrBlank() && authentication != null) {
+      val source = account.substringBefore("/")
+      val username = account.substringAfter("/")
+
+      val azureUser = authentication.principal as UserDetailsImpl
+
+      val user = userService.getMasterUserPersonDetails(username, AuthSource.fromNullableString(source))
+      user.map {
+        // TODO: ensure that the email address matches for the user
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/UserSelectorAuthorizationEndpoint.kt:98]: will do this in next PR

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/service/UserContextService.kt [hunk 7]
```diff
@@ -32,7 +32,7 @@ class UserContextService(
     }
   }
 
-  private fun discoverUsers(loginUser: UserPersonDetails, scopes: Set<String>): List<UserPersonDetails> {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/service/UserContextService.kt:35]: will remove the unnecessary `resolveUser` in next PR as using new method here instead

File: .editorconfig (hunk 5 above)
Comment (mikehalmamoj) [reply .editorconfig:16]: I also vote for `ij_kotlin_call_parameters_wrap = off`.

**Review by JamieDraperUK — APPROVED** [review]
LGTM!

**Pull request merged**